### PR TITLE
Enable building from -proposed with old-fashioned

### PR DESCRIPTION
**Pull request by codyshepherd**
The launchpad-buildd supports a `--proposed` flag, and oftentimes it is desirable to do old-fashioned builds against -proposed. As far as I can tell, enabling this amounts to overriding the sources list (thanks John) and passing the flag down to the `buildlivefs` launchpad-buildd in-target operation.

I've tested these changes by building the xenial azure-base target with bartender. I modified the Makefile in secret sauce to explicitly pass the `--proposed` flag as an `old_fashioned_flags` append. I've verified that the package versions in the resulting manifest are from -proposed where expected.

### CODE REVIEW

File: old-fashioned-image-build [hunk 1]
```diff
@@ -199,11 +204,20 @@ export MIRROR=$(egrep "(archive|ports)" /etc/apt/sources.list|head -1 | \
...
-/usr/share/launchpad-buildd/bin/in-target override-sources-list \
-  --backend=lxd --series=$SERIES --arch=${ARCH} $LIVEFS_NAME \
-    "deb http://$MIRROR/${MIRROR_PATH} $SERIES main restricted universe multiverse" \
-    "deb http://$MIRROR/${MIRROR_PATH} $SERIES-security main restricted universe multiverse" \
-    "deb http://$MIRROR/${MIRROR_PATH} $SERIES-updates main restricted universe multiverse"
+if [ -n "$PROPOSED" ]; then
+  /usr/share/launchpad-buildd/bin/in-target override-sources-list \
+    --backend=lxd --series=$SERIES --arch=${ARCH} $LIVEFS_NAME \
+      "deb http://$MIRROR/${MIRROR_PATH} $SERIES main restricted universe multiverse" \
+      "deb http://$MIRROR/${MIRROR_PATH} $SERIES-proposed main restricted universe multiverse" \
```
Comment (davidkrauser) [inline old-fashioned-image-build:211]: @codyshepherd why is this change necessary?

Comment (codyshepherd) [reply old-fashioned-image-build:211]: To enable, e.g. `bionic-proposed` based on the `$PROPOSED` flag. So that we can use old-fashioned/bartender to build against packages in the -proposed pocket.

Comment (codyshepherd) [reply old-fashioned-image-build:211]: Otherwise packages from the -proposed pocket are not pulled in

Comment (rcj4747) [reply old-fashioned-image-build:211]: This is mixing two different contexts for enabling -proposed and they need to be separate.

This first change is adding the -proposed pocket in the build host container, which would pull in livecd-rootfs and other packages used to build images from -proposed.

The second change to add --proposed to `in-target buildlivefs` would enable --proposed in the images we build.

We might want to do either of these, but we need to control them separately.

**Review by rcj4747 — CHANGES_REQUESTED** [review]
The PR is enabling the -proposed pockets in two different contexts.  This needs to be split so we can control them separately.

File: old-fashioned-image-build [hunk 2]
```diff
@@ -199,11 +204,20 @@ export MIRROR=$(egrep "(archive|ports)" /etc/apt/sources.list|head -1 | \
...
+      "deb http://$MIRROR/${MIRROR_PATH} $SERIES-security main restricted universe multiverse" \
+      "deb http://$MIRROR/${MIRROR_PATH} $SERIES-updates main restricted universe multiverse"
+else
+  /usr/share/launchpad-buildd/bin/in-target override-sources-list \
+    --backend=lxd --series=$SERIES --arch=${ARCH} $LIVEFS_NAME \
+      "deb http://$MIRROR/${MIRROR_PATH} $SERIES main restricted universe multiverse" \
+      "deb http://$MIRROR/${MIRROR_PATH} $SERIES-security main restricted universe multiverse" \
+      "deb http://$MIRROR/${MIRROR_PATH} $SERIES-updates main restricted universe multiverse"
+fi
```
Comment (powersj) [inline old-fashioned-image-build:220]: To avoid the duplicated lines, can you:

1) set a variable to incude the three common lines found in both sources.list
2) if proposed is enabled, add that line to the above variable
3) have one call to in-target

**Review by davidkrauser — APPROVED** [review]

**Review by rcj4747 — APPROVED** [review]
Thanks for the updates Cody
